doc/port_link_training/port-link-training-design.md

```diff
@@ -0,0 +1,336 @@
+
+1. SAI implementation must return error code if any of the above attributes is not supported, swss and syncd must not crash.
+2. SAI implementation must keep port link training disabled by default for backward compatible. As long as swss and SAI keep backward compatible, user need not change anything after this feature is implemented and available in SONiC.
+
+### Configuration and management 
+
+#### CLI Enhancements
+
+A few new CLI commands are designed to support port link training.
+
+##### Config link training mode
+
+```
+Format:
+  config interface link-training <interface_name> <mode>
+
+Arguments:
+  interface_name: name of the interface to be configured. e.g: Ethernet0
+  mode: link training mode, can be either "auto", "on" or "off"
+
+Example:
+  config interface link-training Ethernet0 auto
+  config interface link-training Ethernet0 off
+
+Return:
+  error message if interface_name or mode is invalid otherwise empty
+```
+
+##### Show link training status
+
+A new CLI command will be added to display the port link training status. All data of this command are fetched from **APPL_DB**.
+
+```
+Format:
+  show interfaces link-training status <interface_name>
+
+Arguments:
+  interface_name: optional. Name of the interface to be shown. e.g: Ethernet0. If interface_name is not given, this command shows link training status for all interfaces.
+
+Example:
+  show interfaces link-training status
+  show interfaces link-training status Ethernet0
+
+Return:
+  error message if interface_name is invalid otherwise:
+
+admin@sonic:~$ show interfaces link-training status
+  Interface    LT Oper    LT Admin    LT Failure    LT RxStatus    Oper    Admin
+-----------  ---------  ----------  ------------  -------------  ------  -------
+  Ethernet0         on          on          none        trained      up       up
+  Ethernet8        off         off             -              -    down       up
+ Ethernet16         on        auto          none        trained      up       up
+ Ethernet24        off           -             -              -    down       up
+ Ethernet32        off           -             -              -    down       up
+```
+
+
+#### Config DB Enhancements  
+
+A new field **link_training** will be added to **PORT** table:
+
+	; Defines information for port configuration
+	key                     = PORT|port_name     ; configuration of the port
+	; field                 = value
+	...
+	link_training           = STRING             ; link training configuration
+
+Valid value of the new fields are described below:
+
+- link_training:  
+String value, the administratively specified port link training configuration.
+When "link_training" is not specified, the port link training should be disabled  
+
+ | Mode |     Description                                                 |
+ |:----:|:---------------------------------------------------------------:|
+ | auto | Enable link-training if applicable to the transceiver attached  |
```

Comment:
Please remove the auto mode as discussed

Reply:
Sure, will do, thanks for the reminder